x/ecocredit/marketplace/msg_cancel_sell_order.go

```diff
@@ -23,12 +23,18 @@ func (m MsgCancelSellOrder) GetSignBytes() []byte {
 
 // ValidateBasic does a sanity check on the provided data.
 func (m *MsgCancelSellOrder) ValidateBasic() error {
+	if len(m.Seller) == 0 {
+		return sdkerrors.ErrInvalidRequest.Wrapf("seller cannot be empty")
+	}
+
 	if _, err := sdk.AccAddressFromBech32(m.Seller); err != nil {
-		return sdkerrors.ErrInvalidAddress
+		return sdkerrors.ErrInvalidRequest.Wrapf("seller is not a valid address: %s", err)
```

Comment:
went thru and checked our address validation in other messages, seems like we have a lot of inconsistencies, some just use sdkerrors.Wrap, some wrap the ErrInvalidAddress, some wrap InvalidRequest.. 

i'm leaning towards using `ErrInvalidAddress`, but happy to hear cases for `InvalidRequest`

Reply:
I was interpreting `ErrInvalidAddress` to mean the address itself is invalid. We could use `ErrUnauthorized` here rather than `ErrInvalidRequest` but yea... we don't have clear guidelines on when to use what errors. Open #1147.

Reply:
Whoops. I was thinking something different. I think this case should be `ErrInvalidAddress`.